src/ethereum/networkAdapters/RpcAdapter.ts

```diff
@@ -396,7 +377,11 @@ export class RpcAdapter
       params
     }
 
-    url = this.addRpcApiKey(url)
+    url = addRpcApiKey(
+      this.ethEngine.initOptions,
+      this.ethEngine.currencyInfo.pluginId,
+      url
+    )
```

Comment:
I think I've discovered this to be a bug. We're adding API key to the server urls in the constructor, so the url passed into this call should already include the api key. Make sure this private method is always called with the a `baseUrl` from the RpcAdapter config, and if so, remove this as a separate fix commit.

Reply:
No the constructor requires an `EthereumEngine`, so it won't work.

Reply:
`addRpcApiKey `isn't needed. The api keys are always populated in the `networkAdapterConfig`